source/System.cpp

```diff
@@ -303,6 +304,155 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
+            solarPanelModifiers.push_back(F5BRIGHTNESS);
+        }
+        else if(starName == "star/g0" || starName == "star/g0-old")
+        {
+            ramscoopModifiers.push_back(G0WIND);
+            solarPanelModifiers.push_back(G0BRIGHTNESS);
+        }
+        else if(starName == "star/g5" || starName == "star/g5-old")
+        {
+            ramscoopModifiers.push_back(G5WIND);
+            solarPanelModifiers.push_back(G5BRIGHTNESS);
+        }
+        else if(starName == "star/giant")
+        {
+            ramscoopModifiers.push_back(GIANTWIND);
+            solarPanelModifiers.push_back(GIANTBRIGHTNESS);
+        }
+        else if(starName == "star/k0" || starName == "star/k0-old")
+        {
+            ramscoopModifiers.push_back(K0WIND);
+            solarPanelModifiers.push_back(K0BRIGHTNESS);
+        }
+        else if(starName == "star/k5" || starName == "star/k5-old")
+        {
+            ramscoopModifiers.push_back(K5WIND);
+            solarPanelModifiers.push_back(K5BRIGHTNESS);
+        }
+        else if(starName == "star/m0")
+        {
+            ramscoopModifiers.push_back(M0WIND);
+            solarPanelModifiers.push_back(M0BRIGHTNESS);
+        }else if(starName == "star/m4")
+        {
+            ramscoopModifiers.push_back(M4WIND);
+            solarPanelModifiers.push_back(M4BRIGHTNESS);
+        }else if(starName == "star/m8")
+        {
+            ramscoopModifiers.push_back(M8WIND);
+            solarPanelModifiers.push_back(M8BRIGHTNESS);
+        }
+        else if(starName == "star/nova")
+        {
+            ramscoopModifiers.push_back(NOVAWIND);
+            solarPanelModifiers.push_back(NOVABRIGHTNESS);
+        }
+        else if(starName == "star/wr")
+        {
+            ramscoopModifiers.push_back(WRWIND);
+            solarPanelModifiers.push_back(WRBRIGHTNESS);
+        }
+        else
+        {
+            ramscoopModifiers.push_back(1);
+            solarPanelModifiers.push_back(1);
+            // TOTO(Jeremy) get rid of this message statement.
+            Messages::Add("there is an unacounted for star type!");
+            Messages::Add(starName);
+        }
+        
+        starCount++;
+	}
+	ramscoopModifiers.sort();
+	solarPanelModifiers.sort();
+	
+	// Sum up modifiers (currently using 1/n to normalize the sum).
+	ramscoopEffectiveness = 0;
+	solarPanelEffectiveness = 0;
+	for(int i = 0; i < starCount; i++)
+    {
+        ramscoopEffectiveness += (ramscoopModifiers.back()/(i + 1));
+        solarPanelEffectiveness += (solarPanelModifiers.back()/(i + 1));
+        
+        ramscoopModifiers.pop_back();
+        solarPanelModifiers.pop_back();
+    }
+    
+    // Make sure that the effectiveness modifiers are not game brakeingly low or high.
+    ramscoopEffectiveness = std::max(0.25, ramscoopEffectiveness);
+    solarPanelEffectiveness = std::max(0.5, solarPanelEffectiveness);
+    solarPanelEffectiveness = std::min(2.0, solarPanelEffectiveness);
```

Comment:
> Why have a hardcoded min/max value? This restricts modders.

^^ In my opinion, if people _really_ wanted to make their values game-breakingly OP, it's their copy- let them.